crates/ruff_linter/src/rules/pyflakes/mod.rs

```diff
@@ -771,11 +771,11 @@ mod tests {
             &parsed,
             target_version,
         );
-        messages.sort_by_key(Ranged::start);
+        messages.sort_by_key(|diagnostic| diagnostic.expect_range().start());
```

Comment:
I'm slightly worried by all the `expect_range` calls because we'll very likely start adding diagnostics that don't have a start range (e.g. IOError!). I think it was the right call to use `expect_range` as part of this PR but it would be great if we could make (at least the production code) more forgiving if the source file or range is missing.